Modules/System/Barcode Encoder/src/IBarcodeProvider.Interface.al

```diff
@@ -0,0 +1,61 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary> 
+/// Interface IBarcodeProvider.
+/// The interface connects the correct Barcode Encoding Handler of a specific Barcode Provider
+/// </summary>
+interface IBarcodeProvider
+{
+    /// <summary> 
+    /// Function GetBarcodeEncoder to determine the correct Barcode Encoding handler based at the Enum.
+    /// </summary>
+    /// <param name="iBarcodeEncoder">Interface object IBarcodeEncoder to connect the requested Encoding Handler.</param>
+    /// <param name="UseEncoding">Enum BarcodeEncoder of the reqested Barcode Encoder Handler.</param>
+    procedure GetEncoder(var iBarcodeEncoder: interface IBarcodeEncoder; UseSymbology: Enum BarcodeSymbology)
+
+    /// <summary> 
+    /// Shows if this encoder is implemented as a Barcode Font Encoder
+    /// </summary>
+    /// <param name="iBarcodeEncoder">Parameter of type interface IBarcodeEncoder.</param>
+    /// <param name="UseSymbology">Parameter of type Enum BarcodeSymbology.</param>
+    /// <returns>Return variable "Boolean".</returns>
+    procedure IsFontEncoder(var iBarcodeEncoder: interface IBarcodeEncoder; UseSymbology: Enum BarcodeSymbology): Boolean
+
+    /// <summary> 
+    /// Shows if this encoder is implemeted as a Barcode Image in Base64 format
+    /// </summary>
+    /// <param name="iBarcodeEncoder">Parameter of type interface IBarcodeEncoder.</param>
+    /// <param name="UseSymbology">Parameter of type Enum BarcodeSymbology.</param>
+    /// <returns>Return variable "Boolean".</returns>
+    procedure IsBase64ImageEncoder(var iBarcodeEncoder: interface IBarcodeEncoder; UseSymbology: Enum BarcodeSymbology): Boolean
```

Comment:
We should split the two provider interfaces. Don't mix font encoders with image encoders. This can lead to runtime exceptions. We prefer compile time exceptions ;-)